pkg/api/v1alpha1/cloudstackmachineconfig_types_test.go

```diff
@@ -32,6 +32,18 @@ func TestCloudStackMachineConfigDiskOfferingEqual(t *testing.T) {
 	g.Expect(diskOffering1.Equal(diskOffering2)).To(BeTrue())
 }
 
+func TestCloudStackMachineConfigEmptyDiskOfferingEqual(t *testing.T) {
+	var nilDiskOffering v1alpha1.CloudStackResourceDiskOffering
```

Comment:
```suggestion
	var nilDiskOffering *v1alpha1.CloudStackResourceDiskOffering
```
I think * is needed to make this variable `nil`